# scRNA-Seq-tools


# Overview
This repo has many scripts for different single cell analysis tools. 


## single cell tutorials

* [2018 Analysis of single cell RNA-seq data](https://hemberg-lab.github.io/scRNA.seq.course/index.html)

## Single cell RNA-seq tools

* [Seurat](http://www.satijalab.org/seurat.html) is an R package designed for the analysis and visualization of single cell RNA-seq data. It contains easy-to-use implementations of commonly used analytical techniques, including the identification of highly variable genes, dimensionality reduction (PCA, ICA, t-SNE), standard unsupervised clustering algorithms (density clustering, hierarchical clustering, k-means), and the discovery of differentially expressed genes and markers.
* [Monocle](http://cole-trapnell-lab.github.io/monocle-release/) Differential expression and time-series analysis for single-cell RNA-Seq and qPCR experiments.
* [MAGIC](https://github.com/krishnaswamylab/MAGIC) (Markov Affinity-based Graph Imputation of Cells), is a method for imputing missing values restoring structure of large biological datasets.
* [Linking transcriptional and genetic tumor heterogeneity through allele analysis of single-cell RNA-seq data.](https://www.ncbi.nlm.nih.gov/pubmed/29898899) tool [HoneyBADGER](https://github.com/JEFworks/HoneyBADGER)
* [pagoda](http://hms-dbmi.github.io/scde), a set of statistical methods for analyzing single-cell RNA-seq data, including differential expression analysis (Kharchenko et al.) and pathway and geneset overdispersion analysis


